src/main/java/org/jenkinsci/plugins/stashNotifier/Notifiers/DefaultApacheHttpNotifier.java

```diff
@@ -51,32 +52,34 @@
 import java.security.NoSuchAlgorithmException;
 import java.security.UnrecoverableKeyException;
 
-class DefaultApacheHttpNotifier implements HttpNotifier {
+public class DefaultApacheHttpNotifier implements HttpNotifier {
```

Comment:
What do you think of adding a `DelegatingHttpNotifier` that makes the decision of which notifier to use based on the context? I've found that a simple pattern that makes it easy to retain API compatibility.

```
class DelegatingHttpNotifier implements HttpNotifier {
    private final HttpNotifier standard;
    private final HttpNotifier extended;

    NotificationResult send (context, ...) {
        return context.supportsExtended() ? extended.send(...) : standard.send(...)
    } 
}
```